src/components/home/ukraine-map.tsx

```diff
@@ -0,0 +1,113 @@
+"use client";
+
+import { useCallback, useRef, useState } from "react";
+import { useClickOutside } from "@/hooks/useClickOutside";
+import { useTranslations } from "next-intl";
+import Image from "next/image";
+
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+
+import {
+  distractsList,
+  showBottom,
+  showLeft,
+} from "@/lib/ukraine-districts/districts-list";
+
+export const UkraineMap = () => {
+  const t = useTranslations("Home.search.map");
+
+  const [selectedPart, setSelectedPart] = useState("");
+  const mapRef = useRef<SVGSVGElement>(null);
+
+  const handleClick = useCallback(
```

Comment:
There is no point in this `useCallback` , since when you use it, you make inline arrow function from it, which triggers re-renders anyway. If you really want to memoize this function, you need to remove `selectedDistrict` from arguments, and make useCallback depend on it. Then you can pass memoized fn as `onClick` prop of path without inline arrow function